src/App.tsx

```diff
@@ -17,18 +58,36 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                sortType={setSortBy}
+                sortValue={sortBy}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {(!todos)
+                ? (<Loader />)
+                : (
+                  <TodoList
+                    todoList={todos !== null ? todos : []}
+                    setUserModal={setUserModal}
+                    eyeMark={eyeMark}
+                    setEyeMark={setEyeMark}
+                  />
+                )}
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {check(userModal)
+        && (
+          <TodoModal
```

Comment:
modal should open immediately, and show loading during user fetch

Reply:
we don`t know, what we have to show modal todo or modal error, it depends on getUser, then I show them the result of the request, the user doesn't need to wait, he will see the mark when it is done.